x/accounts/genesis.go

```diff
@@ -0,0 +1,94 @@
+package accounts
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/collections"
+	v1 "cosmossdk.io/x/accounts/v1"
+)
+
+func (k Keeper) ExportState(ctx context.Context) (*v1.GenesisState, error) {
+	genState := &v1.GenesisState{}
+
+	// get account number
+	accountNumber, err := k.AccountNumber.Peek(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	genState.AccountNumber = accountNumber
+
+	err = k.AccountsByType.Walk(ctx, nil, func(key []byte, value string) (stop bool, err error) {
+		accState, err := k.exportAccount(ctx, key, value)
+		if err != nil {
+			return true, err
+		}
+		genState.Accounts = append(genState.Accounts, accState)
+		return false, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return genState, nil
+}
+
+func (k Keeper) exportAccount(ctx context.Context, addr []byte, accType string) (*v1.GenesisState_GenesisAccount, error) {
+	addrString, err := k.addressCodec.BytesToString(addr)
+	if err != nil {
+		return nil, err
+	}
+	account := &v1.GenesisState_GenesisAccount{
+		Address:     addrString,
+		AccountType: accType,
+	}
+	rng := new(collections.Range[[]byte]).
+		Prefix(addr)
+	err = k.AccountsState.Walk(ctx, rng, func(key, value []byte) (stop bool, err error) {
```

Comment:
does accountstate get set anywhere outside genesis?

here would you not be walking over empty account state?